next-gen-ui/final/lib/common/shader_painter.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+
+class ShaderPainter extends CustomPainter {
+  ShaderPainter(this.shader, {this.update});
+
+  final FragmentShader shader;
+  final void Function(FragmentShader, Size)? update;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    update?.call(shader, size);
+    canvas.drawRect(
+      Rect.fromLTWH(0, 0, size.width, size.height),
+      Paint()..shader = shader,
+    );
+  }
+
+  @override
+  bool shouldRepaint(covariant CustomPainter oldDelegate) {
+    return true;
+  }
```

Comment:
since this is for a codelab implement this correctly:

```suggestion
  bool shouldRepaint(covariant ShaderPainter oldDelegate) {
    return oldDelegate.shader != shader || oldDelegate.update != update;
  }
```

Reply:
`shader_painter.dart` deleted